src/Jobs/Feeds/Feed.php

```diff
@@ -163,39 +163,36 @@ private function generate()
             return;
         }
 
-        $expectedProductCount = $query->count();
+        $expectedProductCount = $this->getProductsCount($query);
 
         // Chunk-process the products
         $query->chunkById($this->chunkSize, function ($products) {
+            $feedLines = collect([]);
             // Map the validProducts into feed rows
-            $feedLines = $products
-                ->map(function ($product) {
+            $products
+                ->each(function ($product) use ($feedLines) {
                     try {
                         // Call the productToFeedRow method on the extending class (AdmarktFeed, BeslistFeed, etc).
```

Comment:
I would change this part, because it looks like you call `postProcessFeedRow` for each row if there are multiple rows, but then call it again afterwards. What to you think of:

```
$feedRow = $this->productToFeedRow($product);
$feedRows = Arr::isAssoc($feedRow) ? [$feedRow] : $feedRow;

foreach($feedRows as $row) {
    $feedLines->push($this->postProcessFeedRow($row, $product));
}
```

By wrapping the row in an array before, you call `postProcessFeedRow` once, and have the same logic for one feed row or multiple rows.